pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -283,11 +283,12 @@ def _do_catch(self, pokemon, encounter_id, catch_rate_by_ball, is_vip=False):
         maximum_ball = ITEM_ULTRABALL if is_vip else ITEM_GREATBALL
         ideal_catch_rate_before_throw = 0.9 if is_vip else 0.35
 
-        berry_count = self.bot.item_inventory_count(berry_id)
-        items_stock = self.bot.current_inventory()
-
         while True:
 
+            self.bot.latest_inventory = None
+            berry_count = self.bot.item_inventory_count(berry_id)
+            items_stock = self.bot.current_inventory()
```

Comment:
Do yiu mean that it this case we should manually modify remaining balls/berries amount? Because we need to know exact amount before pokemon catch/retry.